repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -238,54 +239,50 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
       }
     }
 
+    def buildTenantToRolesMap(token: ValidToken): TenantToRolesMap = {
+      token.defaultTenantId.map(_ -> Set.empty[String]).toMap ++
+        token.roles.groupBy(_.tenantId).map({ case (key, value) => key.getOrElse(DomainRoleTenantKey) -> value.map(_.name).toSet })
+    }
+
     def buildTenantHeader(defaultTenant: Option[String],
                           roleTenants: Seq[String],
-                          matchedUriTenant: Option[String]): Vector[String] = {
+                          matchedTenants: Set[String]): Vector[String] = {
       val sendAllTenants = configuration.getTenantHandling.isSendAllTenantIds
       val sendTenantIdQuality = Option(configuration.getTenantHandling.getSendTenantIdQuality)
       val sendQuality = sendTenantIdQuality.isDefined
       val defaultTenantQuality = sendTenantIdQuality.map(_.getDefaultTenantQuality).getOrElse(0.0)
-      val uriTenantQuality = sendTenantIdQuality.map(_.getUriTenantQuality).getOrElse(0.0)
+      val matchedTenantQuality = sendTenantIdQuality.map(_.getUriTenantQuality).getOrElse(0.0)
       val rolesTenantQuality = sendTenantIdQuality.map(_.getRolesTenantQuality).getOrElse(0.0)
 
       case class PreferredTenant(id: String, quality: Double)
 
-      val preferredTenant = (defaultTenant, matchedUriTenant) match {
-        case (Some(default), Some(uri)) =>
-          val quality = if (default.equals(uri)) {
-            math.max(defaultTenantQuality, uriTenantQuality)
-          } else {
-            uriTenantQuality
-          }
-          Some(PreferredTenant(uri, quality))
-        case (None, Some(uri)) =>
-          Some(PreferredTenant(uri, uriTenantQuality))
-        case (Some(default), None) =>
-          Some(PreferredTenant(default, defaultTenantQuality))
-        case (None, None) =>
-          if (roleTenants.nonEmpty) Some(PreferredTenant(roleTenants.head, rolesTenantQuality))
-          else None
-      }
+      val preferredTenants =
+        if (matchedTenants.nonEmpty) {
```

Comment:
I feel like there's a match in here somewhere, but all i can come up with still has these if's

Reply:
I mean, you could do a match, but you would probably end up using guards... which are just ifs anyway. If/elseif/else branches are actually the simpler construct, and one (of a few) that pattern matching uses behind-the-scenes, so if it makes sense, I tend to prefer if/elseif/else.

Reply:
That's what i was saying, every match i could come up with just had the if guards, and actually became less readable. I just feel like there is another version of this that somehow doesn't do that, but i can't find it. This is fine as written.